contrib/k8s.io/client-go/kubernetes/kubernetes.go

```diff
@@ -21,11 +22,18 @@ func WrapRoundTripper(rt http.RoundTripper) http.RoundTripper {
 	return httptrace.WrapRoundTripper(rt,
 		httptrace.WithBefore(func(req *http.Request, span ddtrace.Span) {
 			span.SetTag(ext.ServiceName, "kubernetes")
-			span.SetTag(ext.ResourceName, requestToResource(req.Method, req.URL.Path))
+			span.SetTag(ext.ResourceName, RequestToResource(req.Method, req.URL.Path))
+			traceID := span.Context().TraceID()
+			if traceID == 0 {
+				return
+			}
+			kubeAuditID := strconv.FormatUint(traceID, 10)
+			req.Header.Set("Audit-Id", kubeAuditID)
+			span.SetTag("kubernetes.auditID", kubeAuditID)
 		}))
 }
 
-func requestToResource(method, path string) string {
+func RequestToResource(method, path string) string {
```

Comment:
Why was this made public API? If there's a good reason it's fine, but then let's clearly document it.

Reply:
I would like to use this part of the code to generate traces from the apiserver side while parsing the Kubernetes audit logs. 
It avoids me to repeat this function and keep a coherence there.
The goal of this is to set a distributed trace.
Does it make sense ?
I'll document this properly.

Reply:
Sure. Let's keep it then. Was just making sure this was intentional.